src/components/Filter/Category/CategoryMenuItem.tsx

```diff
@@ -1,37 +1,83 @@
-import React from 'react';
-import { CategoryMenuItemProps } from './category.type';
+import React, { useState, useEffect } from 'react';
+import { ComponentProps } from './category.type';
+import { FilterParams } from '../filter.type';
 
-const CategoryMenuItem: React.FC<CategoryMenuItemProps> = ({
+const CategoryMenuItem: React.FC<ComponentProps> = ({
     label,
     count,
     childrenItems,
+    params,
+    setParams,
 }) => {
+    const [isExpanded, setIsExpanded] = useState(false);
+    const [isSelected, setSelected] = useState(false);
+
+    useEffect(() => {
+        setSelected(params.category === label);
+    }, [params]);
+
+    const toggleExpand = () => {
+        setIsExpanded(!isExpanded);
+    };
+
+    const handleCategorySelect = () => {
+        setParams((prevParams: FilterParams) => ({
+            ...prevParams,
+            category: label,
+        }));
+        setSelected((prevSelected) => {
+            const newState = !prevSelected;
+            console.log(`Category ${label} selected:`, newState);
+            return newState;
```

Comment:
```suggestion
            return !prevSelected;
```
Mình sẽ k push console.log lên e nhé